pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -299,9 +299,14 @@ def dump_checkpoint(self, weights_only: bool = False) -> dict:
             checkpoint['callbacks'] = callback_states
 
             # dump optimizers
-            optimizer_states = []
-            for i, optimizer in enumerate(self.trainer.optimizers):
-                optimizer_states.append(optimizer.state_dict())
+            if self.trainer.accelerator_backend:
+                # Rely on accelerator backend to obtain optimizer state from processes
+                optimizer_states = self.trainer.accelerator_backend.optimizer_states
+            else:
+                optimizer_states = []
+                for i, optimizer in enumerate(self.trainer.optimizers):
+                    optimizer_states.append(optimizer.state_dict())
```

Comment:
should this entire section rely on the trainer's accelerator backend for the optimizer states?

Reply:
TIL we define a cpu accelerator, thought we don't... we should entirely rely on the backend, will modify :)